lib/common/type-utils.ts

```diff
@@ -59,15 +58,23 @@ export function serialize (value: any): any {
 export function deserialize (value: any): any {
   if (value && value.__ELECTRON_SERIALIZED_NativeImage__) {
     const image = nativeImage.createEmpty();
-    for (const rep of value.representations) {
-      const { size, scaleFactor, dataURL } = rep;
+
+    // Use Buffer when there's only one representation for better perf.
+    // This avoids compressing to/from PNG where it's not necessary to
```

Comment:
Interesting in knowing more about this, why can't multiple representations use the bitmap buffer?

Reply:
If one uses a buffer to construct an image with multiple representation from multiple `dataURLs`, the dataURLs themselves are not preserved. If you test:

```
    const image = nativeImage.createEmpty();

    const dataURL1 = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAAC0lEQVQYlWNgAAIAAAUAAdafFs0AAAAASUVORK5CYII=';
    image.addRepresentation({ scaleFactor: 1.0, dataURL: dataURL1 });

    const dataURL2 = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAIAAAACCAIAAAD91JpzAAAAFklEQVQYlWP8//8/AwMDEwMDAwMDAwAkBgMBBMzldwAAAABJRU5ErkJggg==';
    image.addRepresentation({ scaleFactor: 2.0, dataURL: dataURL2 });

    const serializedImage = serialize(image);
    const nonEmpty = deserialize(serializedImage);

    expect(nonEmpty.toDataURL({ scaleFactor: 1.0 })).to.equal(dataURL1);
    expect(nonEmpty.toDataURL({ scaleFactor: 2.0 })).to.equal(dataURL2);
```

`expect(nonEmpty.toDataURL({ scaleFactor: 2.0 })).to.equal(dataURL2);` will fail, as the only `dataURL` associated with the image will be `dataURL1`.